### PR TITLE
Update subnet when host IP changes instead of delete-old + create-new subnet

### PR DESCRIPTION
**Pull request by pravisankar**
closes: https://github.com/openshift/openshift-sdn/issues/307

### CODE REVIEW

**Comment by pravisankar** [comment]
Origin PR: https://github.com/openshift/origin/pull/8879

**Comment by pravisankar** [comment]
@openshift/networking PTAL

File: plugins/osdn/ovs/proxy.go [hunk 1]
```diff
@@ -65,8 +65,12 @@ func (proxy *ovsProxyPlugin) watchPods() {
 
 		switch eventType {
 		case watch.Added, watch.Modified:
+			glog.V(5).Infof("Watch Added or Modified event for Pod %q", pod.ObjectMeta.Name)
```
Comment (danwinship) [inline plugins/osdn/ovs/proxy.go]: any reason to not do `"Watch %s event...", string(eventType)`? (Or even `strings.Title(string(eventType))`)

**Comment by danwinship** [comment]
LGTM

File: plugins/osdn/ovs/proxy.go (hunk 1 above)
Comment (pravisankar) [inline plugins/osdn/ovs/proxy.go]: +1, I'll update

File: plugins/osdn/subnets.go [hunk 2]
```diff
@@ -50,52 +50,40 @@ func (oc *OsdnController) addNode(nodeName string, nodeIP string) error {
...
 	if err != nil {
-		// Release subnet if needed
-		if subnetAllocated {
-			_, ipnet, er := net.ParseCIDR(subnetCIDR)
-			if er != nil {
-				log.Errorf("Failed to release subnet %q for node %q: %v", subnetCIDR, nodeName, er)
-			} else {
-				oc.subnetAllocator.ReleaseNetwork(ipnet)
-			}
+		_, ipnet, er := net.ParseCIDR(subnetCIDR)
```
Comment (dcbw) [inline plugins/osdn/subnets.go]: Couldn't we just use "sn" here?

**Comment by dcbw** [comment]
Small nit, otherwise LGTM

**Comment by pravisankar** [comment]
@dcbw @danwinship  Updated

**Comment by danwinship** [comment]
LGTM but lets get the origin side in first since this depends on it

**Pull request merged**